### PR TITLE
concourse-worker: scripts improvements

### PR DESCRIPTION
**Pull request by sdurrheimer**

### CODE REVIEW

**Review by gaelL — APPROVED** [review]

**Comment by sdurrheimer** [comment]
Cleanup script still doesn't seem ideal.

I got the following logs during testing:
```
root@workers-worker-prod-i-06cb260baa956e26e:/var/lib/concourse# /var/lib/concourse/concourse-worker-cleaner
+ LOG_FILE=/var/log/concourse-worker-cleaner.log
+ :
+ exec
++ tee -a /var/log/concourse-worker-cleaner.log
+ log 'Starting /var/lib/concourse/concourse-worker-cleaner script'
++ date
+ echo 'Wed Mar 16 18:12:57 UTC 2022 workers-worker-prod-i-06cb260baa956e26e Worker-Cleaner[20173]: root Starting /var/lib/concourse/concourse-worker-cleaner script'
Wed Mar 16 18:12:57 UTC 2022 workers-worker-prod-i-06cb260baa956e26e Worker-Cleaner[20173]: root Starting /var/lib/concourse/concourse-worker-cleaner script
+ command -v btrfs
+ btrfs filesystem df /var/lib/concourse/datas
+ concourse_work_dir_btrfs_check=0
+ [[ 0 -eq 0 ]]
+ btrfs subvolume list -a /var/lib/concourse/datas
+ awk '-F ' '{print "/var/lib/concourse/datas/"$9}'
+ xargs btrfs subvolume delete
ERROR: cannot delete '/var/lib/concourse/datas/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume': Directory not empty
ERROR: cannot access subvolume /var/lib/concourse/datas/<FS_TREE>/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/946ec6e8be6ef264550403f9d3313dd27a7afb906850a61771a6bf1bc6fe854f: No such file or directory
ERROR: cannot access subvolume /var/lib/concourse/datas/<FS_TREE>/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/349f9fc5d7c32683496ffcb1638f657dc7fb47c91a262f1079deea6680296aa5: No such file or directory
ERROR: cannot access subvolume /var/lib/concourse/datas/<FS_TREE>/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/5c6ae37b7c82ebf2e0439b2746487180d989bb3f37905d644c0e29d3451d71ac: No such file or directory
ERROR: cannot access subvolume /var/lib/concourse/datas/<FS_TREE>/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/5dbbbdc4e54a761bdd2320c666ca25f7e0ffa1af4dc64ac7ad49f8c919dc608f: No such file or directory
ERROR: cannot access subvolume /var/lib/concourse/datas/<FS_TREE>/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/538c8cdeba71ca1eaa8a91b6dd18ef1f13b4cc6c57eb98b654dab46b59a4f628: No such file or directory
ERROR: cannot access subvolume /var/lib/concourse/datas/<FS_TREE>/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/99dbc9f8d8123ed8fc61deced31b3ff0d368f48bbe727f40d61f3c3c63b56f39: No such file or directory
ERROR: cannot access subvolume /var/lib/concourse/datas/<FS_TREE>/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/b982ffd8b913b831040cd02de07a7c38f8de7e7457fd05b30b91fdb315d6ec81: No such file or directory
ERROR: cannot access subvolume /var/lib/concourse/datas/<FS_TREE>/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/b29fb06c1d543531cb6c280a7f659b7a8a92b24d26f499c0cc755e6b70c753c5: No such file or directory
ERROR: cannot access subvolume /var/lib/concourse/datas/<FS_TREE>/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/a901f07424c0407e792dedde468ce1c6baff68efb0336cb554937550133a456a: No such file or directory
ERROR: cannot access subvolume /var/lib/concourse/datas/<FS_TREE>/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/ef962c249dff98ec5bf9f1b8d352fe6df01217fe980b517d36aa1973b4eb81fb: No such file or directory
ERROR: cannot access subvolume /var/lib/concourse/datas/<FS_TREE>/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/e3d221e293f104085ec69e2adc4daa5d2f42e3c68c62ab6fe6396722d8d5576d: No such file or directory
ERROR: cannot access subvolume /var/lib/concourse/datas/<FS_TREE>/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/01aabe85516c624475976bcf579b3932be17595ac939307725b7ec02f29ace46-init: No such file or directory
ERROR: cannot access subvolume /var/lib/concourse/datas/<FS_TREE>/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/01aabe85516c624475976bcf579b3932be17595ac939307725b7ec02f29ace46: No such file or directory
Delete subvolume (no-commit): '/var/lib/concourse/datas/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume'
+ rm -rf /var/lib/concourse/datas/volumes
rm: cannot remove '/var/lib/concourse/datas/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/946ec6e8be6ef264550403f9d3313dd27a7afb906850a61771a6bf1bc6fe854f': Operation not permitted
rm: cannot remove '/var/lib/concourse/datas/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/349f9fc5d7c32683496ffcb1638f657dc7fb47c91a262f1079deea6680296aa5': Operation not permitted
rm: cannot remove '/var/lib/concourse/datas/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/5c6ae37b7c82ebf2e0439b2746487180d989bb3f37905d644c0e29d3451d71ac': Operation not permitted
rm: cannot remove '/var/lib/concourse/datas/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/5dbbbdc4e54a761bdd2320c666ca25f7e0ffa1af4dc64ac7ad49f8c919dc608f': Operation not permitted
rm: cannot remove '/var/lib/concourse/datas/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/538c8cdeba71ca1eaa8a91b6dd18ef1f13b4cc6c57eb98b654dab46b59a4f628': Operation not permitted
rm: cannot remove '/var/lib/concourse/datas/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/99dbc9f8d8123ed8fc61deced31b3ff0d368f48bbe727f40d61f3c3c63b56f39': Operation not permitted
rm: cannot remove '/var/lib/concourse/datas/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/b982ffd8b913b831040cd02de07a7c38f8de7e7457fd05b30b91fdb315d6ec81': Operation not permitted
rm: cannot remove '/var/lib/concourse/datas/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/b29fb06c1d543531cb6c280a7f659b7a8a92b24d26f499c0cc755e6b70c753c5': Operation not permitted
rm: cannot remove '/var/lib/concourse/datas/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/a901f07424c0407e792dedde468ce1c6baff68efb0336cb554937550133a456a': Operation not permitted
rm: cannot remove '/var/lib/concourse/datas/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/ef962c249dff98ec5bf9f1b8d352fe6df01217fe980b517d36aa1973b4eb81fb': Operation not permitted
rm: cannot remove '/var/lib/concourse/datas/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/e3d221e293f104085ec69e2adc4daa5d2f42e3c68c62ab6fe6396722d8d5576d': Operation not permitted
rm: cannot remove '/var/lib/concourse/datas/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/01aabe85516c624475976bcf579b3932be17595ac939307725b7ec02f29ace46-init': Operation not permitted
rm: cannot remove '/var/lib/concourse/datas/volumes/live/84992f78-b9da-4dcc-40a1-4880ffcd7e1c/volume/docker/btrfs/subvolumes/01aabe85516c624475976bcf579b3932be17595ac939307725b7ec02f29ace46': Operation not permitted
+ log 'Ending /var/lib/concourse/concourse-worker-cleaner script'
++ date
+ echo 'Wed Mar 16 18:12:57 UTC 2022 workers-worker-prod-i-06cb260baa956e26e Worker-Cleaner[20173]: root Ending /var/lib/concourse/concourse-worker-cleaner script'
Wed Mar 16 18:12:57 UTC 2022 workers-worker-prod-i-06cb260baa956e26e Worker-Cleaner[20173]: root Ending /var/lib/concourse/concourse-worker-cleaner script
```

EDIT: improved a bit the command following a bit more testing

File: templates/concourse-worker.service.j2 [hunk 1]
```diff
@@ -4,12 +4,14 @@
 Description=concourse-worker
 Requires=network-online.target
 After=network-online.target
+Wants=concourse-worker-watchdog.service
```
Comment (gaelL) [inline templates/concourse-worker.service.j2:8]: Maybe we could add a if condition in this template, as the watchdog is not always enabled eg https://github.com/cycloid-community-catalog/stack-external-worker/blob/master/ansible/external-worker.yml#L202-L215

**Review by gaelL — CHANGES_REQUESTED** [review]

File: templates/concourse-worker.service.j2 (hunk 1 above)
Comment (sdurrheimer) [reply templates/concourse-worker.service.j2:8]: True, forgot about that.

Comment (sdurrheimer) [reply templates/concourse-worker.service.j2:8]: done

**Review by gaelL — APPROVED** [review]

**Pull request merged**